website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,475 @@
+name label in the `module` block and record the old name inside a `moved` block:
+
+```hcl
+module "b" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+When creating the next plan for each configuration using this module, Terraform
+will treat any existing object addresses beginning with `module.a` as if
+they had instead been created in `module.b`. `module.a.aws_instance.example`
+would be treated as `module.b.aws_instance.example`.
+
+Both of the addresses in this example referred to a module call as a whole, and
+so Terraform recognizes the move for all instances of the call. If this
+module call used `count` or `for_each` then it would apply to all of the
+instances, without the need to specify each one separately.
+
+## Enabling `count` and `for_each` For a Module Call
+
+Imagine that the initial version of your module contains a single-instance
+module:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+Applying this configuration would cause Terraform to create objects whose
+addresses begin with `module.a`.
+
+Later requirements might lead to you needing to use
+[`count`](../meta-arguments/count.html) with this resource, in order
+to systematically declare multiple instances. To preserve an object that
+was previously associated with `aws_instance.a` alone, you can used a `moved`
+block to specify which instance key that object will take in the new
+configuration:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+  count  = 3
+
+  # (module arguments)
+}
+
+moved {
+  from = module.a
+  to   = module.a[2]
+}
+```
+
+The above will cause Terraform to treat all objects in `module.a` as if they
+were originally created in `module.a[2]`, and thus it will plan to create
+new objects only for `module.a[0]` and `module.a[1]`.
+
+When at least one of the two addresses includes an instance key, like
+`[2]` in the above example, Terraform will understand both addresses as
+referring to specific _instances_ of a module call rather than the module
+call as a whole. That means you can use `moved` to switch between keys and to
+add and remove keys as you switch between `count`, `for_each`, or neither.
+
+For more examples of recording moves associated with instances, refer to
+the similar section
+[Enabling `count` and `for_each` For a Resource](#enabling-count-and-for_each-for-a-resource).
+
+# Splitting One Module into Multiple
+
+As a module grows to support new requirements, it might eventually grow big
+enough to warrant splitting into two separate modules.
+
+Consider the following initial version of a module:
+
+```hcl
+resource "aws_instance" "a" {
+  # (other resource-type-specific configuration)
+}
+
+resource "aws_instance" "b" {
+  # (other resource-type-specific configuration)
+}
+
+resource "aws_instance" "c" {
+  # (other resource-type-specific configuration)
+}
+```
+
+Imagine that we intend to split this into two modules as follows:
```

Comment:
```suggestion
You can split this into two modules as follows:
```

Reply:
One thing I didn't really like about my first draft was the various places where I wrote "Imagine that...", or similar. That's not a sentence structure we typically use, but I was struggling for some other way to introduce a hypothetical situation before showing how to address that situation.

I know your feedback here is about this specific case, and this one is a little different than the other examples anyway, but I'd love to hear if you have any ideas on how to restructure the other "Imagine that..." sentences here.

Reply:
Thank you for asking! I added in some additional comments to try to address this. Generally I think yes - we don't want to say "imagine" though I completely understand why you initially did that. It can be hard to do this without phrases like "Let's say" or "Imagine that". But, I think actually that it's just fine to say "Consider this example module:" [show example] "Later, you split this into two modules and add a block....."   I think this way you're making it active, concise, and still walking the user through the example in a way that makes sense. I think this also works because before this section you tell users that they're about to see a bunch of example walkthroughs. THoughts?

Reply:
Thanks! I've incorporated all of your suggestions below about "Imagine that..." sentences, and I previously incorporated the one I replied to here, so I'm going to resolve this conversation to represent that.